xbmc/addons/GUIViewStateAddonBrowser.cpp

```diff
@@ -40,6 +41,15 @@ CGUIViewStateAddonBrowser::CGUIViewStateAddonBrowser(const CFileItemList& items)
   else
   {
     AddSortMethod(SortByLabel, SortAttributeIgnoreFolders, 551, LABEL_MASKS("%L", "%I", "%L", ""));  // Filename, Size | Foldername, empty
+
+    if (URIUtils::PathStarts(items.GetPath(), "addons://sources/"))
```

Comment:
Toggling through the new filters is not working in each category. Video addons seem fine, but ```addons://sources/audio/``` for e.g. breaks if you change to get last used and try to get back to by title.
It just stays as it was sorted last, without updating ever again